File: banchus-common/src/main/java/pe/nanamochi/banchus/util/Privileges.java
package pe.nanamochi.banchus.util;

import pe.nanamochi.banchus.domain.enums.ClientPrivileges;
import pe.nanamochi.banchus.domain.enums.ServerPrivileges;

public class Privileges {
  private Privileges() {}

  public static int serverToClientPrivileges(int priv) {
    int ret = 0;
    if ((priv & ServerPrivileges.SUPPORTER.getValue()) != 0) {
      ret |= ClientPrivileges.SUPPORTER.getValue();
    }
    if ((priv & ServerPrivileges.CHAT_MODERATOR.getValue()) != 0) {
      ret |= ClientPrivileges.MODERATOR.getValue();
    }
    if ((priv & ServerPrivileges.SUPER_ADMIN.getValue()) != 0) {
      ret |= ClientPrivileges.OWNER.getValue();
    }
    return ret;
  }
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/util/Privileges.kt
package pe.nanamochi.banchus.util

import pe.nanamochi.banchus.domain.enums.ClientPrivileges
import pe.nanamochi.banchus.domain.enums.ServerPrivileges

fun Int.toClientPrivileges(): Int {
    var ret = 0

    if (this and ServerPrivileges.SUPPORTER.value != 0) {
        ret = ret or ClientPrivileges.SUPPORTER.value
    }

    if (this and ServerPrivileges.CHAT_MODERATOR.value != 0) {
        ret = ret or ClientPrivileges.MODERATOR.value
    }

    if (this and ServerPrivileges.SUPER_ADMIN.value != 0) {
        ret = ret or ClientPrivileges.OWNER.value
    }

    return ret
}
